### ISSUE TITLE
Database connection

### ISSUE THREAD

**Issue by itzamnamx** [fsmn/narrative#13]
Hi,

I am trying to setup the application, but  I got the message

Fatal error: Class 'CI_DB_mysql_result' not found in /home/villaeducativa/public_html/ingenian.mx/narrative/system/database/DB_driver.php on line 715

I am using a database.php file with this content

<?php  if ( ! defined('BASEPATH')) exit('No direct script access allowed');

$active_group = 'default';
$db['default']['hostname'] = "localhost";
$db['default']['username'] = "dbuser";
$db['default']['password'] = "dbpassword";
$db['default']['database'] = "dbnarrative";
$db['default']['dbdriver'] = "mysql";
$db['default']['dbprefix'] = "";
$db['default']['pconnect'] = FALSE;
$db['default']['db_debug'] = FALSE;
$db['default']['cache_on'] = FALSE;
$db['default']['cachedir'] = "";
$db['default']['char_set'] = "utf8";
$db['default']['dbcollat'] = "utf8_general_ci";
$db['default']['swap_pre'] = "";
$db['default']['autoinit'] = TRUE;
$db['default']['stricton'] = FALSE;
$db['default']['port'] = 3306;

**Comment by fsmn** [comment]
I typically use mysqli as the driver. Do you have that available? I haven't
tried this with standard mysql and some of my queries may rely on mysqli
structured queries.

On Mon, Apr 11, 2016 at 12:55 AM, Itzamna <notifications@github.com> wrote:

> Hi,
>
> I am trying to setup the application, but I got the message
>
> Fatal error: Class 'CI_DB_mysql_result' not found in
> /home/villaeducativa/public_html/
> ingenian.mx/narrative/system/database/DB_driver.php on line 715
>
> I am using a database.php file with this content
>
> <?php if ( ! defined('BASEPATH')) exit('No direct script access allowed');
>
> $active_group = 'default';
> $db['default']['hostname'] = "localhost";
> $db['default']['username'] = "dbuser";
> $db['default']['password'] = "dbpassword";
> $db['default']['database'] = "dbnarrative";
> $db['default']['dbdriver'] = "mysql";
> $db['default']['dbprefix'] = "";
> $db['default']['pconnect'] = FALSE;
> $db['default']['db_debug'] = FALSE;
> $db['default']['cache_on'] = FALSE;
> $db['default']['cachedir'] = "";
> $db['default']['char_set'] = "utf8";
> $db['default']['dbcollat'] = "utf8_general_ci";
> $db['default']['swap_pre'] = "";
> $db['default']['autoinit'] = TRUE;
> $db['default']['stricton'] = FALSE;
> $db['default']['port'] = 3306;
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/fsmn/narrative/issues/13>
>



-- 
Technology Director
Friends School of Minnesota
1365 Englewood Ave. Saint Paul, MN 55104
direct: 651-621-8954
tel: 651-917-0636
fax: 651-917-0708
http://www.fsmn.org

Friends School of Minnesota prepares children to embrace life, learning,
and
community with hope, skill, understanding and creativity. We are committed
to the Quaker values of peace, justice, simplicity and integrity.

**Comment by fsmn** [comment]
I am also happy to help you get started once you get past this initial
setup. Here's the latest schema in case the one on the github is out of
date. Also attached is the initial admin user account. For reasons that
would be difficult to explain, and that I hope to fix in newer versions,
the admin user is not the user with ID 1.

Chris

On Mon, Apr 11, 2016 at 3:01 PM, Chris Dart <chrisd@fsmn.org> wrote:

> I typically use mysqli as the driver. Do you have that available? I
> haven't tried this with standard mysql and some of my queries may rely on
> mysqli structured queries.
>
> On Mon, Apr 11, 2016 at 12:55 AM, Itzamna <notifications@github.com>
> wrote:
>
>> Hi,
>>
>> I am trying to setup the application, but I got the message
>>
>> Fatal error: Class 'CI_DB_mysql_result' not found in
>> /home/villaeducativa/public_html/
>> ingenian.mx/narrative/system/database/DB_driver.php on line 715
>>
>> I am using a database.php file with this content
>>
>> <?php if ( ! defined('BASEPATH')) exit('No direct script access allowed');
>>
>> $active_group = 'default';
>> $db['default']['hostname'] = "localhost";
>> $db['default']['username'] = "dbuser";
>> $db['default']['password'] = "dbpassword";
>> $db['default']['database'] = "dbnarrative";
>> $db['default']['dbdriver'] = "mysql";
>> $db['default']['dbprefix'] = "";
>> $db['default']['pconnect'] = FALSE;
>> $db['default']['db_debug'] = FALSE;
>> $db['default']['cache_on'] = FALSE;
>> $db['default']['cachedir'] = "";
>> $db['default']['char_set'] = "utf8";
>> $db['default']['dbcollat'] = "utf8_general_ci";
>> $db['default']['swap_pre'] = "";
>> $db['default']['autoinit'] = TRUE;
>> $db['default']['stricton'] = FALSE;
>> $db['default']['port'] = 3306;
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/fsmn/narrative/issues/13>
>>
>
>
>
> --
> Technology Director
> Friends School of Minnesota
> 1365 Englewood Ave. Saint Paul, MN 55104
> direct: 651-621-8954
> tel: 651-917-0636
> fax: 651-917-0708
> http://www.fsmn.org
>
> Friends School of Minnesota prepares children to embrace life, learning,
> and
> community with hope, skill, understanding and creativity. We are committed
> to the Quaker values of peace, justice, simplicity and integrity.
>



-- 
Technology Director
Friends School of Minnesota
1365 Englewood Ave. Saint Paul, MN 55104
direct: 651-621-8954
tel: 651-917-0636
fax: 651-917-0708
http://www.fsmn.org

Friends School of Minnesota prepares children to embrace life, learning,
and
community with hope, skill, understanding and creativity. We are committed
to the Quaker values of peace, justice, simplicity and integrity.

**Comment by fsmn** [comment]
I have tested this system with mysql and it does return an error. Not as
fatal as the one you are using. What version of mysql are you running? I
have only tested this on version 5.0.11 and higher.

On Mon, Apr 11, 2016 at 3:34 PM, Chris Dart <chrisd@fsmn.org> wrote:

> I am also happy to help you get started once you get past this initial
> setup. Here's the latest schema in case the one on the github is out of
> date. Also attached is the initial admin user account. For reasons that
> would be difficult to explain, and that I hope to fix in newer versions,
> the admin user is not the user with ID 1.
>
> Chris
>
> On Mon, Apr 11, 2016 at 3:01 PM, Chris Dart <chrisd@fsmn.org> wrote:
>
>> I typically use mysqli as the driver. Do you have that available? I
>> haven't tried this with standard mysql and some of my queries may rely on
>> mysqli structured queries.
>>
>> On Mon, Apr 11, 2016 at 12:55 AM, Itzamna <notifications@github.com>
>> wrote:
>>
>>> Hi,
>>>
>>> I am trying to setup the application, but I got the message
>>>
>>> Fatal error: Class 'CI_DB_mysql_result' not found in
>>> /home/villaeducativa/public_html/
>>> ingenian.mx/narrative/system/database/DB_driver.php on line 715
>>>
>>> I am using a database.php file with this content
>>>
>>> <?php if ( ! defined('BASEPATH')) exit('No direct script access
>>> allowed');
>>>
>>> $active_group = 'default';
>>> $db['default']['hostname'] = "localhost";
>>> $db['default']['username'] = "dbuser";
>>> $db['default']['password'] = "dbpassword";
>>> $db['default']['database'] = "dbnarrative";
>>> $db['default']['dbdriver'] = "mysql";
>>> $db['default']['dbprefix'] = "";
>>> $db['default']['pconnect'] = FALSE;
>>> $db['default']['db_debug'] = FALSE;
>>> $db['default']['cache_on'] = FALSE;
>>> $db['default']['cachedir'] = "";
>>> $db['default']['char_set'] = "utf8";
>>> $db['default']['dbcollat'] = "utf8_general_ci";
>>> $db['default']['swap_pre'] = "";
>>> $db['default']['autoinit'] = TRUE;
>>> $db['default']['stricton'] = FALSE;
>>> $db['default']['port'] = 3306;
>>>
>>> —
>>> You are receiving this because you are subscribed to this thread.
>>> Reply to this email directly or view it on GitHub
>>> <https://github.com/fsmn/narrative/issues/13>
>>>
>>
>>
>>
>> --
>> Technology Director
>> Friends School of Minnesota
>> 1365 Englewood Ave. Saint Paul, MN 55104
>> direct: 651-621-8954
>> tel: 651-917-0636
>> fax: 651-917-0708
>> http://www.fsmn.org
>>
>> Friends School of Minnesota prepares children to embrace life, learning,
>> and
>> community with hope, skill, understanding and creativity. We are
>> committed
>> to the Quaker values of peace, justice, simplicity and integrity.
>>
>
>
>
> --
> Technology Director
> Friends School of Minnesota
> 1365 Englewood Ave. Saint Paul, MN 55104
> direct: 651-621-8954
> tel: 651-917-0636
> fax: 651-917-0708
> http://www.fsmn.org
>
> Friends School of Minnesota prepares children to embrace life, learning,
> and
> community with hope, skill, understanding and creativity. We are committed
> to the Quaker values of peace, justice, simplicity and integrity.
>



-- 
Technology Director
Friends School of Minnesota
1365 Englewood Ave. Saint Paul, MN 55104
direct: 651-621-8954
tel: 651-917-0636
fax: 651-917-0708
http://www.fsmn.org

Friends School of Minnesota prepares children to embrace life, learning,
and
community with hope, skill, understanding and creativity. We are committed
to the Quaker values of peace, justice, simplicity and integrity.

**Comment by itzamnamx** [comment]
I have tried that with the GoDaddy site 5.5.45-cll-lve - MySQL Community Server (GPL) and PHP version 5.6. I will try with the new schema. And let you know. thank you!

**Comment by fsmn** [comment]
I am happy to help. Make sure to switch the msql to mysqli. It will work
better. the mysql driver is deprecated.

On Tue, Apr 12, 2016 at 6:59 PM, Itzamna <notifications@github.com> wrote:

> I have tried that with the GoDaddy site 5.5.45-cll-lve - MySQL Community
> Server (GPL) and PHP version 5.6. I will try with the new schema. And let
> you know. thank you!
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/fsmn/narrative/issues/13#issuecomment-209156306>
>



-- 
Technology Director
Friends School of Minnesota
1365 Englewood Ave. Saint Paul, MN 55104
direct: 651-621-8954
tel: 651-917-0636
fax: 651-917-0708
http://www.fsmn.org

Friends School of Minnesota prepares children to embrace life, learning,
and
community with hope, skill, understanding and creativity. We are committed
to the Quaker values of peace, justice, simplicity and integrity.

**Comment by itzamnamx** [comment]
<?php  if ( ! defined('BASEPATH')) exit('No direct script access allowed');
/*
| -------------------------------------------------------------------
| DATABASE CONNECTIVITY SETTINGS
| -------------------------------------------------------------------
| This file will contain the settings needed to access your database.
|
| For complete instructions please consult the 'Database Connection'
| page of the User Guide.
|
| -------------------------------------------------------------------
| EXPLANATION OF VARIABLES
| -------------------------------------------------------------------
|
|   ['hostname'] The hostname of your database server.
|   ['username'] The username used to connect to the database
|   ['password'] The password used to connect to the database
|   ['database'] The name of the database you want to connect to
|   ['dbdriver'] The database type. ie: mysql.  Currently supported:
                 mysql, mysqli, postgre, odbc, mssql, sqlite, oci8
|   ['dbprefix'] You can add an optional prefix, which will be added
|                to the table name when using the  Active Record class
|   ['pconnect'] TRUE/FALSE - Whether to use a persistent connection
|   ['db_debug'] TRUE/FALSE - Whether database errors should be displayed.
|   ['cache_on'] TRUE/FALSE - Enables/disables query caching
|   ['cachedir'] The path to the folder where cache files should be stored
|   ['char_set'] The character set used in communicating with the database
|   ['dbcollat'] The character collation used in communicating with the database
|                NOTE: For MySQL and MySQLi databases, this setting is only used
|                as a backup if your server is running PHP < 5.2.3 or MySQL < 5.0.7
|                (and in table creation queries made with DB Forge).
|                There is an incompatibility in PHP with mysql_real_escape_string() which
|                can make your site vulnerable to SQL injection if you are using a
|                multi-byte character set and are running versions lower than these.
|                Sites using Latin-1 or UTF-8 database character set and collation are unaffected.
|   ['swap_pre'] A default table prefix that should be swapped with the dbprefix
|   ['autoinit'] Whether or not to automatically initialize the database.
|   ['stricton'] TRUE/FALSE - forces 'Strict Mode' connections
|                           - good for ensuring strict SQL while developing
|
| The $active_group variable lets you choose which connection group to
| make active.  By default there is only one group (the 'default' group).
|
| The $active_record variables lets you determine whether or not to load
| the active record class
*/

$active_group = 'default';
$db['default']['hostname'] = "localhost";
$db['default']['username'] = "palosolo";
$db['default']['password'] = "gusana02";
$db['default']['database'] = "narrative";
$db['default']['dbdriver'] = "mysqli";
$db['default']['dbprefix'] = "";
$db['default']['pconnect'] = FALSE;
$db['default']['db_debug'] = FALSE;
$db['default']['cache_on'] = FALSE;
$db['default']['cachedir'] = "";
$db['default']['char_set'] = "utf8";
$db['default']['dbcollat'] = "utf8_general_ci";
$db['default']['swap_pre'] = "";
$db['default']['autoinit'] = TRUE;
$db['default']['stricton'] = FALSE;
$db['default']['port'] = 3306;

And I got this 

Fatal error: Call to a member function row() on a non-object in /home/villaeducativa/public_html/ingenian.mx/narrative/system/libraries/Session/drivers/Session_database_driver.php on line 160

URL is available at

http://ingenian.mx/narrative/

I am checking which other setup could cause this issue.

**Comment by fsmn** [comment]
Ok. This has to do with a script expecting data when there is none. I can
create some insert queries with arbitrary data you can work with.

On Tue, Apr 12, 2016 at 7:08 PM, Itzamna <notifications@github.com> wrote:

> <?php if ( ! defined('BASEPATH')) exit('No direct script access allowed');
> /*
> | -------------------------------------------------------------------
> | DATABASE CONNECTIVITY SETTINGS
> | -------------------------------------------------------------------
> | This file will contain the settings needed to access your database.
> |
> | For complete instructions please consult the 'Database Connection'
> | page of the User Guide.
> |
> | -------------------------------------------------------------------
> | EXPLANATION OF VARIABLES
> | -------------------------------------------------------------------
> |
> | ['hostname'] The hostname of your database server.
> | ['username'] The username used to connect to the database
> | ['password'] The password used to connect to the database
> | ['database'] The name of the database you want to connect to
> | ['dbdriver'] The database type. ie: mysql. Currently supported:
> mysql, mysqli, postgre, odbc, mssql, sqlite, oci8
> | ['dbprefix'] You can add an optional prefix, which will be added
> | to the table name when using the Active Record class
> | ['pconnect'] TRUE/FALSE - Whether to use a persistent connection
> | ['db_debug'] TRUE/FALSE - Whether database errors should be displayed.
> | ['cache_on'] TRUE/FALSE - Enables/disables query caching
> | ['cachedir'] The path to the folder where cache files should be stored
> | ['char_set'] The character set used in communicating with the database
> | ['dbcollat'] The character collation used in communicating with the
> database
> | NOTE: For MySQL and MySQLi databases, this setting is only used
> | as a backup if your server is running PHP < 5.2.3 or MySQL < 5.0.7
> | (and in table creation queries made with DB Forge).
> | There is an incompatibility in PHP with mysql_real_escape_string() which
> | can make your site vulnerable to SQL injection if you are using a
> | multi-byte character set and are running versions lower than these.
> | Sites using Latin-1 or UTF-8 database character set and collation are
> unaffected.
> | ['swap_pre'] A default table prefix that should be swapped with the
> dbprefix
> | ['autoinit'] Whether or not to automatically initialize the database.
> | ['stricton'] TRUE/FALSE - forces 'Strict Mode' connections
> | - good for ensuring strict SQL while developing
> |
> | The $active_group variable lets you choose which connection group to
> | make active. By default there is only one group (the 'default' group).
> |
> | The $active_record variables lets you determine whether or not to load
> | the active record class
> */
>
> $active_group = 'default';
> $db['default']['hostname'] = "localhost";
> $db['default']['username'] = "palosolo";
> $db['default']['password'] = "gusana02";
> $db['default']['database'] = "narrative";
> $db['default']['dbdriver'] = "mysqli";
> $db['default']['dbprefix'] = "";
> $db['default']['pconnect'] = FALSE;
> $db['default']['db_debug'] = FALSE;
> $db['default']['cache_on'] = FALSE;
> $db['default']['cachedir'] = "";
> $db['default']['char_set'] = "utf8";
> $db['default']['dbcollat'] = "utf8_general_ci";
> $db['default']['swap_pre'] = "";
> $db['default']['autoinit'] = TRUE;
> $db['default']['stricton'] = FALSE;
> $db['default']['port'] = 3306;
>
> And I got this
>
> Fatal error: Call to a member function row() on a non-object in
> /home/villaeducativa/public_html/
> ingenian.mx/narrative/system/libraries/Session/drivers/Session_database_driver.php
> on line 160
>
> URL is available at
>
> http://ingenian.mx/narrative/
>
> I am checking which other setup could cause this issue.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/fsmn/narrative/issues/13#issuecomment-209157996>
>



-- 
Technology Director
Friends School of Minnesota
1365 Englewood Ave. Saint Paul, MN 55104
direct: 651-621-8954
tel: 651-917-0636
fax: 651-917-0708
http://www.fsmn.org

Friends School of Minnesota prepares children to embrace life, learning,
and
community with hope, skill, understanding and creativity. We are committed
to the Quaker values of peace, justice, simplicity and integrity.

**Comment by itzamnamx** [comment]
Ok, that would help a lot.

**Comment by fsmn** [comment]
Here you go. This will replace the teacher table contents. You may want to
delete that row from this file.

There are a number of site-wide settings that I have entered into the
database over time that do not have a user-interface. Let me know what ones
there are and I can try to add them during my next major update.

Chris

On Tue, Apr 12, 2016 at 7:27 PM, Itzamna <notifications@github.com> wrote:

> Ok, that would help a lot.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/fsmn/narrative/issues/13#issuecomment-209163441>
>



-- 
Technology Director
Friends School of Minnesota
1365 Englewood Ave. Saint Paul, MN 55104
direct: 651-621-8954
tel: 651-917-0636
fax: 651-917-0708
http://www.fsmn.org

Friends School of Minnesota prepares children to embrace life, learning,
and
community with hope, skill, understanding and creativity. We are committed
to the Quaker values of peace, justice, simplicity and integrity.

**Comment by itzamnamx** [comment]
It works! well now send me to a auth page in fsmn I will continue to check how to redirect it.

: )

Victor

**Comment by fsmn** [comment]
Make sure to edit the config.php to match your domain.

On Tue, Apr 12, 2016 at 8:28 PM, Itzamna <notifications@github.com> wrote:

> It works! well now send me to a auth page in fsmn I will continue to check
> how to redirect it.
>
> : )
>
> Victor
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/fsmn/narrative/issues/13#issuecomment-209180578>
>



-- 
Technology Director
Friends School of Minnesota
1365 Englewood Ave. Saint Paul, MN 55104
direct: 651-621-8954
tel: 651-917-0636
fax: 651-917-0708
http://www.fsmn.org

Friends School of Minnesota prepares children to embrace life, learning,
and
community with hope, skill, understanding and creativity. We are committed
to the Quaker values of peace, justice, simplicity and integrity.

**Comment by itzamnamx** [comment]
Tahnk you Chris, is is working, now is pointing to the auth but I got this

Parse error: syntax error, unexpected '}' in /home/villaeducativa/public_html/ingenian.mx/application/views/auth/login.php on line 5

A PHP Error was encountered

Severity: Parsing Error

Message: syntax error, unexpected '}'

Filename: auth/login.php

Line Number: 5

Backtrace

**Comment by itzamnamx** [comment]
There is one blank space in the login.php file, line 5

NON WORKING
$classes = " errors";
WORKING
$classes = "errors";

**Comment by itzamnamx** [comment]
After that it ask for a password, which is retrieved using the hashing I sorted it out changing to my valid email and then, reseting the password to get a valid one. I think we can close this thread and open a new one about initial values. 
Also I have checked this module that can help to install the application

https://github.com/mikecrittenden/codeigniter-installer

I will give a try and then let you know.

Thank you.
